README.md

```diff
@@ -11,6 +11,7 @@ The operator manages:
 
 ## Documentation
 
+- [Compatibility & Support docs](./docs/compatibility.md)
```

Comment:
If we're adding this, I think we should also remove https://github.com/open-telemetry/opentelemetry-operator?tab=readme-ov-file#compatibility